src/openfermion/utils/_pubchem.py

```diff
@@ -11,43 +11,50 @@
 #   limitations under the License.
 
 
-def geometry_from_pubchem(name):
+def geometry_from_pubchem(name, structure=None):
     """Function to extract geometry using the molecule's name from the PubChem
-    database.
+    database. The 'structure' argument can be used to specify which structure
+    info to use to extract the geometry. If structure=None, the geometry will
+    be constructed based on 3D info, if available, otherwise on 2D (to keep
+    backwards compatibility with the times when the argument 'structure'
+    was not implemented).
 
     Args:
         name: a string giving the molecule's name as required by the PubChem
             database.
+        structure: a string '2d' or '3d', to specify a specific structure
+            information to be retrieved from pubchem. The default is None.
+            Recommended value is '3d'.
 
     Returns:
         geometry: a list of tuples giving the coordinates of each atom with
         distances in Angstrom.
     """
     import pubchempy
 
-    pubchempy_2d_molecule = pubchempy.get_compounds(name, 'name',
-                                                    record_type='2d')
-
-    # Check if 2-D geometry is available. If not then no geometry is.
-    if not pubchempy_2d_molecule:
-        print('Unable to find molecule in the PubChem database.')
+    if structure in ['2d', '3d']:
+        pubchempy_molecule = pubchempy.get_compounds(name, 'name',
+                                                     record_type=structure)
+    elif structure is None:
+        # Ideally get the 3-D geometry if available.
+        pubchempy_molecule = pubchempy.get_compounds(name, 'name',
+                                                        record_type='3d')
+
+        # If the 3-D geometry isn't available, get the 2-D geometry instead.
+        if not pubchempy_molecule:
+            pubchempy_molecule = pubchempy.get_compounds(name, 'name',
+                                                        record_type='2d')
+    else:
+        raise ValueError('Incorrect value for the argument structure=%s' % structure)
```

Comment:
It seems this line is not covered by a test.  You can construct this test by mocking a get_compounds method from pubchempy which returns something other than ['2d', '3d', None] and then checking the ValueError with pytest.raises.

Reply:
On second thought...I'll just merge this now.  Coveralls doesn't seem to mind.   I'll add a test to this later.